fp16 related build issues on JLSE

```
[100%] Built target RecursiveGaussian
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/fp16_math.dir/half_math.cpp.o: in function `__half::operator float<float, (void*)0>() const':
/home/pvelesko/t/chip-spv/include/hip/spirv_hip_fp16.h:296: undefined reference to `__extendhfsf2'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/fp16_math.dir/half_math.cpp.o: in function `__half<float, nullptr>':
/home/pvelesko/t/chip-spv/include/hip/spirv_hip_fp16.h:167: undefined reference to `__truncsfhf2'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/fp16_math.dir/half_math.cpp.o: in function `__half<short, nullptr>':
/home/pvelesko/t/chip-spv/include/hip/spirv_hip_fp16.h:181: undefined reference to `__truncsfhf2'
/usr/lib64/gcc/x86_64-suse-linux/7/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/fp16_math.dir/half_math.cpp.o: in function `__half::operator unsigned short<unsigned short, (void*)0>() const':
/home/pvelesko/t/chip-spv/include/hip/spirv_hip_fp16.h:310: undefined reference to `__extendhfsf2'
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,7 @@
 
 # temporary
 set(CMAKE_WARN_DEPRECATED OFF)
-add_compile_options(-Wno-format-extra-args)
+add_compile_options(-Wno-format-extra-args -mf16c)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-duplicate-decl-specifier \
 -Wno-tautological-constant-compare  -Wno-c++20-extensions -Wno-unused-result \

